kayenta-core/src/main/java/com/netflix/kayenta/config/KayentaConfiguration.java

```diff
@@ -64,4 +79,37 @@ MetricSetMixerService metricSetMixerService() {
   StorageServiceRepository storageServiceRepository() {
     return new MapBackedStorageServiceRepository();
   }
+
+  //
+  // Configure subtypes of CanaryMetricSetQueryConfig, all of which must be
+  // defined in the package com.netflix.kayenta.canary.providers
+  //
+  @Bean public ObjectMapperSubtypeConfigurer.ClassSubtypeLocator assetSpecSubTypeLocator() {
+    return new ObjectMapperSubtypeConfigurer.ClassSubtypeLocator(CanaryMetricSetQueryConfig.class, ImmutableList.of("com.netflix.kayenta.canary.providers"));
```

Comment:
I wonder if it makes sense to put these in `com.netflix.kayenta.canary.providers.metricsservices` or something similar. Thinking it may be nicer in the future when we may be doing something similar for judges or other components.

Reply:
Good point.  I'll see about doing this before I merge.